src/Magnum/Platform/Sdl2Application.cpp

```diff
@@ -275,6 +275,52 @@ void Sdl2Application::setWindowTitle(const std::string& title) {
     #endif
 }
 
+#ifndef CORRADE_TARGET_EMSCRIPTEN
+void Sdl2Application::setCursor(Cursor cursor) {
+    CORRADE_INTERNAL_ASSERT(UnsignedInt(cursor) < Containers::arraySize(_cursors));
+
+    _cursor = cursor;
+
+    if(cursor == Cursor::Hidden) {
+        SDL_ShowCursor(SDL_FALSE);
+        return;
+    } else if(cursor == Cursor::HiddenLocked) {
+        SDL_SetWindowGrab(_window, SDL_TRUE);
+        SDL_SetRelativeMouseMode(SDL_TRUE);
+        return;
+    } else {
+        SDL_ShowCursor(SDL_TRUE);
+        SDL_SetWindowGrab(_window, SDL_FALSE);
+        SDL_SetRelativeMouseMode(SDL_FALSE);
+    }
+
+    if(!_cursors[UnsignedInt(cursor)]) {
+        constexpr SDL_SystemCursor CursorMap[] {
+            SDL_SYSTEM_CURSOR_ARROW,
+            SDL_SYSTEM_CURSOR_IBEAM,
+            SDL_SYSTEM_CURSOR_WAIT,
+            SDL_SYSTEM_CURSOR_CROSSHAIR,
+            SDL_SYSTEM_CURSOR_WAITARROW,
+            SDL_SYSTEM_CURSOR_SIZENWSE,
+            SDL_SYSTEM_CURSOR_SIZENESW,
+            SDL_SYSTEM_CURSOR_SIZEWE,
+            SDL_SYSTEM_CURSOR_SIZENS,
+            SDL_SYSTEM_CURSOR_SIZEALL,
+            SDL_SYSTEM_CURSOR_NO,
+            SDL_SYSTEM_CURSOR_HAND
+        };
+
+        _cursors[UnsignedInt(cursor)] = SDL_CreateSystemCursor(CursorMap[UnsignedInt(cursor)]);
+    }
+
+    SDL_SetCursor(_cursors[UnsignedInt(cursor)]);
+}
+
+Sdl2Application::Cursor Sdl2Application::cursor() {
+    return _cursor;
```

Comment:
Instead of using a variable, this could be a query to `SDL_GetCursor()` (and `SDL_GetRelativeMouseMode()`), I think? Especially considering the interaction with the to-be-deprecated `setMouseLocked()` APIs that could enable relative mouse mode independently, the getter should return the real state and not what it *thinks* is active. This would just iterate over the `_cursor` array, trying to find an item that matches (and then, if nothing matches, `CORRADE_ASSERT_UNREACHABLE()`, at least for now).

Tried to find an equivalent getter for GLFW, but there doesn't seems to be (unless I overlooked some general `glfwGetValue()`), so there the variable is okay I guess.

Reply:
True, that's better.

> (and then, if nothing matches, CORRADE_ASSERT_UNREACHABLE(), at least for now)

Wouldn't it better to return Cursor::Arrow as default? Because if you query the cursor without having created one it would just crash, but I think that it's obvious that the default one is always the Cursor::Arrow.
A minor downside is that if someone wants to manage the cursor by itself by calling internal SDL functions, the cursor() always returns Cursor::Arrow instead of saying "Hey, there's nothing here", but I think that this cloud just be specified in the documentation.

Reply:
> Because if you query the cursor without having created one it would just crash

Good point. Yeah, do that then :) 

> if someone wants to manage the cursor by itself by calling internal SDL functions

Yeah, that was my primary concern, didn't realize it would crash also if the user didn't set any cursor yet.